metrics/prometheus.go

```diff
@@ -1212,6 +1213,36 @@ func (c *PrometheusCollector) collectContainersInfo(ch chan<- prometheus.Metric)
 			}
 		}
 	}
+
+	if c.includedMetrics.Has(container.AppMetrics) {
+		for _, container := range containers {
+			cstats := container.Stats
+			if len(cstats) > 0 {
+				last := cstats[len(cstats)-1]
+				for metricLabel, v := range last.CustomMetrics {
+					for _, metric := range v {
+						values := make([]string, 0, len(metric.Labels)+2)
+						labels := make([]string, 0, len(metric.Labels)+2)
+						labels = append(labels, "container_name")
+						values = append(values, container.ContainerReference.Name)
+						alias := strings.Join(container.ContainerReference.Aliases, ".")
+						labels = append(labels, "container_alias")
+						values = append(values, alias)
+						for label, value := range metric.Labels {
+							if label == "__name__" {
+								continue
+							}
+							labels = append(labels, sanitizeLabelName(label))
```

Comment:
Keep in mind that all metrics returned in a single scrape must have the same labels.  See https://github.com/google/cadvisor/pull/2262/files#diff-88eab3cc8cef9ad727beea9923056cbbR1148-R1172 above.  I believe this would cause the metrics endpoint to return only some metrics on each scrape.